staging/src/k8s.io/client-go/rest/request_test.go

```diff
@@ -2474,7 +2485,7 @@ func TestRequestWithRetry(t *testing.T) {
 			name:                         "server returns retryable err, request body Seek returns error, retry aborted",
 			body:                         &readSeeker{err: io.EOF},
 			serverReturns:                responseErr{response: nil, err: io.ErrUnexpectedEOF},
-			errExpected:                  io.ErrUnexpectedEOF,
+			errContains:                  "failed to reset request body for *rest.Request - EOF",
```

Comment:
Is this the right error to return?  Why would the caller want to deal with the wrapper?

Reply:
This test is very non-obvious why it fails.  Technically the test is saying "i have an empty body", it's just that Seek now returns EOF, which is abnormal.

As far as the message, this message is probably not accurate.  The programmer has given us a body that is not reusable for retries, which is programmer error OR a transient error.  We probably can't tell the difference easily, so the message needs to reflect that.

I would probably suggest the error message be changed to `failed to reset the request body while retrying a request: EOF`